WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteFragment.kt

```diff
@@ -151,6 +171,62 @@ class MySiteFragment : Fragment(R.layout.my_site_fragment),
         viewModel.setActionableEmptyViewGone(actionableEmptyView.isVisible) {
             actionableEmptyView.setVisible(false)
         }
+        siteInfo.loadMySiteDetails(state.siteInfo)
+        showAppbar()
+    }
+
+    private fun MySiteInfoCardBinding.loadMySiteDetails(site: SiteInfoCard) {
+        siteTitle = site.title
+        if (site.iconState is IconState.Visible) {
+            mySiteBlavatar.visibility = View.VISIBLE
+            imageManager.load(mySiteBlavatar, BLAVATAR, site.iconState.url ?: "")
+            mySiteIconProgress.visibility = View.GONE
+            mySiteBlavatar.setOnClickListener { site.onIconClick.click() }
+        } else if (site.iconState is IconState.Progress) {
+            mySiteBlavatar.setOnClickListener(null)
+            mySiteIconProgress.visibility = View.VISIBLE
+            mySiteBlavatar.visibility = View.GONE
+        }
+        quickStartIconFocusPoint.setVisibleOrGone(site.showIconFocusPoint)
+        if (site.onTitleClick != null) {
+            siteInfoContainer.title.setOnClickListener { site.onTitleClick.click() }
+        } else {
+            siteInfoContainer.title.setOnClickListener(null)
+        }
+        siteInfoContainer.title.text = site.title
+        quickStartTitleFocusPoint.setVisibleOrGone(site.showTitleFocusPoint)
+        siteInfoContainer.subtitle.text = site.url
+        siteInfoContainer.subtitle.setOnClickListener { site.onUrlClick.click() }
+        switchSite.setOnClickListener { site.onSwitchSiteClick.click() }
+    }
+
+    private fun MySiteFragmentBinding.showAppbar() {
+        header.setVisible(true)
+        if (viewModel.isMySiteTabsEnabled) showSiteInfoToolbarWithTabs()
+        else showSiteInfoToolbarWithoutTabs()
+    }
+
+    private fun MySiteFragmentBinding.showSiteInfoToolbarWithTabs() {
+        val newHeight = resources.getDimension(R.dimen.app_bar_with_site_info_tabs_height).toInt()
+        appbarMain.layoutParams.height = newHeight
+        updateToolbarBottomMargin(newHeight)
+        tabLayout.setVisible(true)
+        appbarMain.requestLayout()
+    }
+
+    private fun MySiteFragmentBinding.showSiteInfoToolbarWithoutTabs() {
+        val newHeight = resources.getDimension(R.dimen.app_bar_with_site_info_height).toInt()
+        appbarMain.layoutParams.height = newHeight
+        updateToolbarBottomMargin(0)
+        tabLayout.setVisible(false)
+        appbarMain.requestLayout()
+    }
```

Comment:
@ashiagr This is a good suggestion.  I have updated the `ViewModel` with a class `SiteInfoToolbarView` which holds the necessary logic for updating the AppBar in "site selected"/ "no site" / "tabs feature enabled" state. 

> This should also help add tests for SiteInfoToolbarView visibility with and without tabs.
Earlier I had visibility logic in `showSiteInfoToolbarWithTabs`, I found that these were not necessary as only the height needs to be changed. So I removed those. Since there is no visibility logic associated with `SiteInfoToolbarView` I have not added any tests for it. It contains only `DimenRes`. 

Let me know what do you think.  You can find the changes in 998c0fb. 🤔

Reply:
In my sleep, naming the class which holds the dimension for the siteInfoToolbarView as `SiteInfoToolbarView` haunted me, so I renamed it first thing in the morning. Sorry for that mistake. 😄 . I have renamed it to `SiteInfoToolbarViewParams`. You can find the changes at 80434e8. Let me know if you have a better name or any other suggestions.

Reply:
No worries @AjeshRPai, changes look much better. 👍 Thank you!